lib/index.js

```diff
@@ -79,20 +79,30 @@ PostHTML.prototype.process = function (tree, options) {
    * @prop {?Boolean} options.sync - enables sync mode, plugins will run synchronously, throws an error when used with async plugins
    * @prop {?Function} options.parser - use custom parser, replaces default (posthtml-parser)
    * @prop {?Function} options.render - use custom render, replaces default (posthtml-render)
-   * @prop {?Boolean} options.skipParse - disable parsing
    */
   options = options || {}
 
+  if (typeof tree !== 'string' || typeof tree !== 'object') {
```

Comment:
Yep proper Tree Validator API  (`validate(tree)` && `tree.validate(node)`) would be better of course, for the first I'm not sure how and it would likely be perf hell :), but authors should be able to validate the transformed nodes. That could be prototyped in e.g`posthtml-plugin-utils` first but a bit off-topic here :)

```js
const $ = require('posthtml-plugin-utils')
...
tree.walk((node) => {
  node = $(node)
    ...
    .class('name', () => {}) 
    .children(() => {}) // typeof node.conten[x] === 'object' 
    .validate()

  return node 
})